pkg/bicep/schematobicep.go

```diff
@@ -36,16 +39,79 @@ func SchemaToBicep(in io.Reader) ([]byte, error) {
 }
 
 func createBicepParameter(name string, sch *schema.Schema, buf *bytes.Buffer) error {
-	bicepType, err := getBicepType(sch.Type)
+	bicepType, err := getBicepTypeFromSchema(sch.Type)
 	if err != nil {
 		return err
 	}
 
-	buf.WriteString(fmt.Sprintf("param %s %s\n", name, bicepType))
+	writeDescription(sch, buf)
+	writeAllowedParams(sch, buf)
+	writeMinValue(sch, buf, bicepType)
+	writeMaxValue(sch, buf, bicepType)
+	writeMinLength(sch, buf, bicepType)
+	writeMaxLength(sch, buf, bicepType)
+	writeSecure(sch, buf, bicepType)
+	writeBicepParam(name, sch, buf, bicepType)
 	return nil
 }
 
-func getBicepType(schemaType string) (string, error) {
+func writeBicepParam(name string, sch *schema.Schema, buf *bytes.Buffer, bicepType string) error {
+	var defVal string
+
+	if sch.Default != nil {
+		renderedVal, err := renderBicep(sch.Default, "")
+		if err != nil {
+			return err
+		}
+
+		defVal = fmt.Sprintf(" = %s", renderedVal)
+	}
+
+	buf.WriteString(fmt.Sprintf("param %s %s%s\n", name, bicepType, defVal))
+	return nil
+}
+
+func renderBicep(val interface{}, prefix string) (string, error) {
+	interfaceType, err := getBicepTypeFromInterface(val)
+	if err != nil {
+		return "", err
+	}
+
+	switch interfaceType {
+	case "string":
+		return fmt.Sprintf("'%s'", val), nil
+	case "int", "bool":
+		return fmt.Sprintf("%v", val), nil
+	case "array":
+		assertedVal, asserArrErr := val.([]interface{})
+		if asserArrErr != true {
+			return "", errors.New("failed to assert array type")
```

Comment:
This error is a bit ambiguous. If it was a large file, it would be hard to tell what it failed to parse. You can use `fmt.Errorf()` to make this a bit more clear.

```suggestion
			return "", fmt.Errorf("unable to convert value into array: %v", val)
```